src/util/property.h

```diff
@@ -39,19 +39,32 @@ class Property {
         previous_ = current_;
     }
     /* Setters */
-    void Set(const T& value, bool makeDirty = true) {
+    bool Set(const T& value, bool makeDirty = true) {
+        if (value == current_) {
+            previous_ = value;
```

Comment:
What is the need to set `previous_` here, why not just return false right away?

Reply:
There's no particular _need_ to do that but in case the past value was different, this sets them to be the same.